app/controllers/ui_job_wizard_controller.rb

```diff
@@ -57,4 +57,19 @@ def resources
     render :json => { :results =>
       resource_list.sort_by { |r| r[:name] }.take(100), :subtotal => resource_list.count}
   end
+
+  def job_invocation
+    job = JobInvocation.authorized.find(params[:id])
+    composer = JobInvocationComposer.from_job_invocation(job, params).params
+    job_template_inputs = JobTemplate.authorized.find(composer[:template_invocations][0][:template_id]).template_inputs_with_foreign
+    inputs = Hash[job_template_inputs.map{ |input| ["inputs[#{input[:name]}]", (composer[:template_invocations][0][:input_values].select {|value| value[:template_input_id]== input[:id]})[0][:value]] } ]
```

Comment:
I'd do something like this. At least to me it reads a bit better.

```suggestion
    pattern_template_invocation = composer[:template_invocations].first
    job_template_inputs = JobTemplate.authorized.find(pattern_template_invocation[:template_id]).template_inputs_with_foreign
    inputs = job_template_inputs.reduce({}) do |acc, input|
      value = pattern_template_invocation[:input_values].find { |value| value[:template_input_id] == input[:id] }
      acc.merge("inputs[#{input[:name]}]" => value[:value])
    end

```

Reply:
To re-use the autofilling from the url which comes in this format. I can change it but I think since we already have a function that does the same, to re-use it. 
autofill pr for context: https://github.com/theforeman/foreman_remote_execution/pull/629

Reply:
Ah, that makes sense. Let's keep it as is then